python/pyspark/mllib/linalg.py

```diff
@@ -670,6 +668,23 @@ def toArray(self):
         """
         return self.values.reshape((self.numRows, self.numCols), order='F')
 
+    def toSparse(self):
+        """Convert to Sparse Format"""
+        numRows = self.numRows
```

Comment:
This will fail if the last (few) columns are only zeros. I have made a minor change to this. Thanks!